### PR TITLE
Systest 9305

### PR DESCRIPTION
**Pull request by kdivya153**

### CODE REVIEW

File: cli/src/config.mjs [hunk 1]
```diff
@@ -21,15 +21,16 @@
...
 //   server directory/sub-repo and the value in the .mf-SAMPLE.config.json file also in
 //   this directory/sub-repo.
 
 const config = {
   app: {
     socketPort: 9998,
     httpPort: 3333,
-    defaultUserId: '12345'
+    defaultUserId: '12345',
+    defaultMerged: 'true'
```
Comment (kschrief) [inline cli/src/config.mjs:33]: We should define the "default" merged state as true in the API definition. We don't have to add it as a configuration.

**Review by kschrief — CHANGES_REQUESTED** [review]

File: cli/src/usage.mjs [hunk 2]
```diff
@@ -23,6 +23,7 @@
 const lines = [
   { cmdInfo: "--help",                                                                    comment: "Show this helpful usage information" },
   { cmdInfo: "--user <userId> ...(see below)...",                                         comment: "Target the given user; Overrides the .mf.config.json file, if present" },
+  { cmdInfo: "--merge",                                                                   comment: "Shows the state of the user; includes global/group state if merge true" },
```
[the remark this answers is not in the document]
Comment (kdivya153) [reply cli/src/usage.mjs]: Addressed

File: cli/src/config.mjs (hunk 1 above)
Comment (kdivya153) [reply cli/src/config.mjs]: Removed from config

File: server/src/config.mjs [hunk 3]
```diff
@@ -42,6 +42,7 @@ const config = {
     developerToolPort: 9995,            // Port for Firebolt to connect to
     developerToolName: 'Mock Firebolt', // Used when publishing with DNS-SD
     defaultUserId: '12345',
+    defaultMerged: 'true',
```
[the remark this answers is not in the document]
Comment (kdivya153) [reply server/src/config.mjs]: Removed from config

File: server/src/routes/api/state.mjs [hunk 4]
```diff
@@ -30,9 +30,12 @@ import * as events from '../../events.mjs';
 
 // --- Route Handlers ---
 
+// GET /api/v1/state
+// params: userId,merged
 function getState(req, res) {
   const userId = getUserIdFromReq(req);
-  const state = stateManagement.getState(userId);
+  const merged =  getMergedFromReq(req);
```
[the remark this answers is not in the document]
Comment (kdivya153) [reply server/src/routes/api/state.mjs]: Addressed

File: server/src/stateManagement.mjs [hunk 5]
```diff
@@ -180,11 +180,12 @@ function getUserId(userId){
 
 // return state based on hierarchy (From lowest priority to highest) global->group->user if mergedState=true
 //return state of the UserId if mergedState=false,to update the state
-function getState(userId,mergedState = true) {
+function getState(userId,mergedState = 'true') {
```
[the remark this answers is not in the document]
Comment (kdivya153) [reply server/src/stateManagement.mjs]: Addressed

File: server/src/stateManagement.mjs [hunk 6]
```diff
@@ -180,11 +180,12 @@ function getUserId(userId){
...
+
   userId = "" + userId;
 //to get the userId from given user/appId
   userId = getUserId(userId);
-  if(!mergedState){
+  if(mergedState != 'true'){
```
[the remark this answers is not in the document]
Comment (kdivya153) [reply server/src/stateManagement.mjs]: Addressed

File: server/src/stateManagement.mjs [hunk 7]
```diff
@@ -587,26 +589,36 @@ function updateState(userId, newState, scope = "") {
     userState = getState(scope,false);
   }
   if ( userState.isDefaultUserState ) {
+    scopeLevel="user"
     if ( scope === config.app.defaultUserId ) {
-      logger.info(`Updating state for default user ${scope}`);
+      logger.info(`Updating state for default user ${scope} and scopeLevel ${scopeLevel}`);
```
[the remark this answers is not in the document]
Comment (kdivya153) [reply server/src/stateManagement.mjs]: Addressed

File: server/test/suite/config.test.mjs [hunk 8]
```diff
@@ -35,6 +35,7 @@ test(`config works properly`, () => {
       developerToolPort: 9995,
       developerToolName: 'Mock Firebolt',
       defaultUserId: '12345',
+      defaultMerged: 'true',
```
[the remark this answers is not in the document]
Comment (kdivya153) [reply server/test/suite/config.test.mjs]: Removed

**Review by kschrief — APPROVED** [review]

**Review by jbigel — APPROVED** [review]

**Pull request merged**